src/data/challenge.ts

```diff
@@ -340,6 +344,15 @@ export abstract class Challenge {
     return false;
   }
 
+  /**
+   * An apply function for TRICK_ROOM challenges. Derived classes should alter this.
+   * @param isTrickRoom {@link Utils.BooleanHolder} Whether the battle is under the effect of Trick Room.
+   * @returns {@link boolean} Whether this function did anything.
```

Comment:
paramNIT:
The type is already given by typescript. not need for redundancy here.
It's a little different at the top with the `@param` (don't ask please)
```suggestion
   * @returns always `false`
```